db/migrate/20220408093837_update_enterprise_social_links.rb

```diff
@@ -0,0 +1,14 @@
+class UpdateEnterpriseSocialLinks < ActiveRecord::Migration
+  INSTAGRAM_REGEX = %r{\A(?:https?://)?(?:www\.)?instagram\.com/([a-zA-Z0-9._@-]{1,30})/?\z}.freeze
+  def up
+      say_with_time "Updating enterprises..." do
+        count = 0
+        Enterprise.where("instagram IS NOT NULL").find_each do |enterprise|
+          say "Update enterprise: #{enterprise.id}"
+          enterprise[:instagram] = enterprise[:instagram].downcase.try(:gsub, INSTAGRAM_REGEX, '\1').delete('@')
```

Comment:
You are changing the attribute in memory but you don't save the record.
```suggestion
          enterprise.update!(instagram: enterprise.instagram.downcase.gsub(INSTAGRAM_REGEX, '\1').delete('@')
```

I'm not sure why you used `try`. If it wasn't a string then `downcase` would have failed already.